chiselFrontend/src/main/scala/chisel3/core/BlackBox.scala

```diff
@@ -52,22 +130,29 @@ case class RawParam(value: String) extends Param
   * }}}
   * @note The parameters API is experimental and may change
   */
-abstract class BlackBox(val params: Map[String, Param] = Map.empty[String, Param]) extends Module {
+abstract class BlackBox(val params: Map[String, Param] = Map.empty[String, Param]) extends BaseBlackBox {
+  def io: Record
```

Comment:
I'm trying to preserve the user-facing behavior of chisel3.Module / chisel3.core.LegacyModule. The main changes are the inclusion of new options like chisel3.MultiIOModule and chisel3.RawModule, which are new APIs.